src/test/java/ProfileTests.java

```diff
@@ -0,0 +1,33 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import pages.HomePage;
+import pages.LoginPage;
+import pages.ProfilePage;
+
+public class ProfileTests extends BaseTest {
+
+    @Test
+    public void changeProfileNameTest() throws InterruptedException {
+        LoginPage loginPage = new LoginPage(getDriver());
+        HomePage homePage = new HomePage(driver);
+        ProfilePage profilePage = new ProfilePage(driver);
+        loginPage.provideEmail("supattra.tangsombutpaiboon@testpro.io").providePassword("te$t$tudent1").clickSubmit();
+
+        homePage.chooseAvatarIcon();
+
+        String randomName = generateRandomName();
+        profilePage.enterCurrentPassword("te$t$tudent1");
+        provideProfileName(randomName);
+        profilePage.clickSaveBtn();
+
+        Thread.sleep(5000);
```

Comment:
Please, delete `throws InterruptedException`

Thread. sleep is bad! It blocks the current thread and renders it unusable for further work.
Using Thread. sleep() frequently in an automation framework is not a good practice.

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.